druid/src/env.rs

```diff
@@ -342,13 +342,20 @@ impl Env {
     ) -> Result<(), ValueTypeError> {
         let env = Arc::make_mut(&mut self.0);
         let key = key.into();
-        // TODO: use of Entry might be more efficient
-        if let Some(existing) = env.map.get(&key) {
-            if !existing.is_same_type(&raw) {
-                return Err(ValueTypeError::new(any::type_name::<V>(), raw));
+        let entry = env.map.entry(key);
+        match entry {
```

Comment:
I'd inline the `entry` variable.
```suggestion
        match env.map.entry(key) {
```